gcapi/create_strategies.py

```diff
@@ -0,0 +1,455 @@
+        ):
+            self.content = yield from self.hybrid_client_api.images.detail(
+                api_url=self.source.image
+            )
+        else:
+            self.content = clean_file_source(self.source)
+
+    def __call__(self):
+        post_request = yield from super().__call__()
+
+        if isinstance(self.content, HyperlinkedImage):
+            # Reuse the existing image
+            post_request.image = self.content.api_url
+            return post_request
+
+        # Upload the image
+        if self.parent is None:
+            # No need to specify target
+            upload_session_data = {}
+        elif isinstance(self.parent, (DisplaySet, DisplaySetPost)):
+            upload_session_data = {
+                "display_set": self.parent.pk,
+                "interface": self.socket.slug,
+            }
+        elif isinstance(self.parent, (ArchiveItem, ArchiveItemPost)):
+            upload_session_data = {
+                "archive_item": self.parent.pk,
+                "interface": self.socket.slug,
+            }
+        else:
+            raise NotImplementedError(
+                f"{type(self.parent)} not supported for uploading images"
+            )
+
+        uploads = []
+        for file in self.content:
+            with open(file, "rb") as f:
+                uploads.append(
+                    (
+                        yield from self.hybrid_client_api.uploads.upload_fileobj(
+                            fileobj=f, filename=file.name
+                        )
+                    )
+                )
+        raw_image_upload_session = (
+            yield from self.hybrid_client_api.raw_image_upload_sessions.create(
+                uploads=[u.api_url for u in uploads],
+                **upload_session_data,
+            )
+        )
+
+        if self.parent is None:
+            post_request.upload_session = raw_image_upload_session.api_url
+            return post_request
+        else:
+            return Empty
+
+
+class ValueSocketValueCreateStrategy(SocketValueCreateStrategy):
+
+    def prepare(self):
+        yield from super().prepare()
+
+        if isinstance(self.source, HyperlinkedComponentInterfaceValue):
+            self._prepare_from_existing_socket_value(socket_value=self.source)
+        else:
+            try:
+                self._prepare_from_local_file(file=self.source)
+            except FileNotFoundError:
+                self._prepare_from_provided_value(value=self.source)
+
+    def _prepare_from_local_file(self, file):
+        cleaned_sources = clean_file_source(file, maximum_number=1)
+        self.content = cleaned_sources[0]
+
+        # Test load
+        with open(self.content) as fp:
+            json.load(fp)
+
+    def _prepare_from_existing_socket_value(self, socket_value):
+        socket = socket_value.interface
+        if socket.super_kind.casefold() == "file":
+            pass  # Nothing to do here, but we accept it
```

Comment:
Maybe I misunderstood something, but why do we accept file type interfaces to be handled by the `ValueSocketCreateStrategy`?

Reply:
Oh no, sorry, pre-first-coffee brain. So this is when someone submits an existing CIV value of a different socket (which could be a file type socket). I would have expected the code to download the file here rather than in `__call__`. 

But in general, isn't it usually a red flag if someone tries to submit socket A value to socket B value? Most of the time, there are schemas attached to our json sockets, and content for socket A is unlikely to be valid for socket B, right?

Reply:
Maybe a good guard-rail to have: only allow socket-to-socket transfers. I think that might be wise to add.

Reply:
Done! Now one can only do exact socket-to-socket copying. Also makes the sourcing of that simpler since we can assume things to be of the exact same kind.